src/main/java/org/jasig/portlet/newsreader/adapter/RomeAdapter.java

```diff
@@ -198,22 +213,57 @@ public void init() throws Exception {
         // second field.
         compressingClient = new DecompressingHttpClient(httpClient);
     }
-
+    
+    protected String[] getPolicyPref(String... policyKey) {
+        String[] values = new String[policyKey.length];
+        
+        try {
+        
+            RequestAttributes requestAttributes = RequestContextHolder.currentRequestAttributes();
+            if (requestAttributes instanceof PortletRequestAttributes) {
+                PortletRequest request = ((PortletRequestAttributes) requestAttributes).getRequest();
+                PortletPreferences prefs = request.getPreferences();
+                List<String> valueList = new ArrayList<String>();
+                for (String key : policyKey) {
+                    valueList.add(prefs.getValue(key, defaultAntiSamyPolicy));
+                }
+                return valueList.toArray(values);
+            } 
+            
+            
+        } catch (IllegalStateException ex) {
+            
+            log.warn("Not Request Bound", ex);
+            
+        }
+        
+        
+        Arrays.fill(values, defaultAntiSamyPolicy);
+        return values;
+    }
+    
     /* (non-Javadoc)
       * @see org.jasig.portlet.newsreader.adapter.INewsAdapter#getSyndFeed(org.jasig.portlet.newsreader.NewsConfiguration, javax.portlet.PortletRequest)
       */
-    public NewsFeed getSyndFeed(NewsConfiguration config, PortletRequest request) throws NewsException {
+    @Override
+    public NewsFeed getSyndFeed(NewsConfiguration config, int page) throws NewsException {
 
-        NewsFeed feed = null;
+        PaginatingNewsFeed feed = null;
 
+        
+        
         // Look for an alternative AntiSamy policy file in the portlet preferences. If found, use it
         // otherwise use the default policyFile being injected into this class via Spring.
         // Note that a policy file string includes a path starting at the application context.
         // (e.g. /WEB-INF/antisamy/antisamy-manchester.xml)
-        PortletPreferences prefs = request.getPreferences();
-        String titlePolicy = prefs.getValue( "titlePolicy", "antisamy-textonly");
-        String descriptionPolicy = prefs.getValue( "descriptionPolicy", "antisamy-textonly");
-
+//        PortletPreferences prefs = request.getPreferences();
+//        String titlePolicy = prefs.getValue( "titlePolicy", "antisamy-textonly");
+//        String descriptionPolicy = prefs.getValue( "descriptionPolicy", "antisamy-textonly");
+        
+        String policy[] = getPolicyPref("titlePolicy", "descriptionPolicy");
```

Comment:
Convert these Magic Numbers into (documented) constants;  confirm they're listed in portlet.xml (for visibility).